simpeg/electromagnetics/natural_source/receivers.py

```diff
@@ -12,320 +14,76 @@ def _alpha(src):
-            'CCVz', 'cell_centers_z'  -> z-component of vector field defined on cell centers
-
-    field : {"e", "h"}
-        Whether to project electric or magnetic fields from the mesh.
-    is_tipper_bs : bool
-        Whether the projection is for a remote base station location where horizontal
-        magnetic fields are measured. Ensures stash names for projection matrices
-        aren't reused.
-
-    Returns
-    -------
-    sp.sparse.csr_matrix
-        The projection matrix.
+class BaseNaturalSourceRx(BaseRx):
     """
-    if mesh.dim < 3:
-        return rx.getP(mesh, projected_grid)
-
-    if is_tipper_bs:
-        stash_name = projected_grid + "_bs"
-    else:
-        stash_name = projected_grid
-
-    if (mesh, stash_name) in rx._Ps:
-        return rx._Ps[(mesh, stash_name, field)]
-
-    if field == "e":
-        locs = rx.locations_e
-    elif field == "h":
-        if isinstance(rx, ApparentConductivity):
-            locs = rx.locations_h
-        elif is_tipper_bs:
-            locs = rx.locations_bs
-        else:
-            locs = rx.locations
-    else:
-        raise ValueError(f"Field type {field} unrecognized. Use 'e' or 'h'")
-
-    P = mesh.get_interpolation_matrix(locs, projected_grid)
-    if rx.storeProjections:
-        rx._Ps[(mesh, stash_name, field)] = P
-    return P
-
-
-class ApparentConductivity(BaseRx):
-    r"""Receiver class for simulating apparent conductivity data (3D problems only).
-
-    This class is used to simulate apparent conductivity data, in S/m, as defined by:
-
-    .. math::
-        \sigma_{app} = \mu_0 \omega \dfrac{\big | \vec{H} \big |^2}{\big | \vec{E} \big |^2}
-
-    where :math:`\omega` is the angular frequency in rad/s,
-
-    .. math::
-        \big | \vec{H} \big | = \Big [ H_x^2 + H_y^2 + H_z^2 \Big ]^{1/2}
-
-    and
-
-    .. math::
-        \big | \vec{H} \big | = \Big [ H_x^2 + H_y^2 + H_z^2 \Big ]^{1/2}
+    Base Class for Natural Source receivers
 
     Parameters
     ----------
-    locations : (n_loc, n_dim) numpy.ndarray
-        Locations where the electric and magnetic fields are measured.
-        If electric and magnetic fields are measured in different locations, please
-        use `locations_e` and `locations_h` when instantiating.
-    locations_e : (n_loc, n_dim) numpy.ndarray
-        Locations where the horizontal electric fields are measured.
-        Must be same size as `locations_h`.
-    locations_h : (n_loc, n_dim) numpy.ndarray
-        Locations where the magnetic fields are measured.
-        Must be same size as `locations_e`.
+    locations : (2) tuple of (n_loc, n_dim) array_like
+        Locations where the two fields are measured.
+        If it is a 3D array, `locations[0]` are the locations of the first field
+        measurements and `locations[1]` are the locations of the second field measurements.
+        Otherwise, they are assumed to be at the same location for a 2D array input.
```

Comment:
The init for BaseNaturalSourceRx has input arguments 'locations1' and 'locations2'. It doesn't look like the docstrings match. I saw that you overrode the setting for 'locations', but that is an input argument for BaseRx.